src/lib/flpTokenService.ts

```diff
@@ -0,0 +1,180 @@
+import { aoInstance } from "@/lib/aoconnect";
+import { redis } from "./redis";
+import { retryWithDelay } from "@/utils/retry.utils";
+
+interface RawFlpToken {
+  flp_token_name: string;
+  flp_token_ticker: string;
+  flp_token_denomination: number;
+  flp_token_logo: string;
+  flp_token_process: string;
+  flp_id: string;
+}
+
+interface FlpToken {
+  name: string;
+  ticker: string;
+  denomination: number;
+  logo: string;
+  id: string;
+  flpId: string;
+  autoClaim: boolean;
+}
+
+type DelegationRecord = Record<string, number>;
+
+interface CachedFlpTokenData {
+  flpTokens: FlpToken[];
+  timestamp: number;
+}
+
+/**
+ * Test token FLP IDs that should be filtered out from the results.
+ * These are test tokens that should not appear in the results.
+ */
+const TEST_TOKEN_FLP_IDS = new Set([
+  "T3M4QSF7VGa0le7KtxBDHOaIcjnZeC-SQ7nh3ABuufs",
+  "So2HpldZaaVFbeH8mUGGzQBVdEzAx5HvMyMaZ47az_M",
+  "4mowY7A-b6WJyVR-Tde2m3Zcl_JVxil21c15PXiHhfA",
+  "-ntvNGm4onpKXS8SZ6-5sFnmjRHfMNAwS_JuR-pO504",
+  "WRkDu1hOeNksAlli1R4LUUh674Q79DjSOSegdIiI68U",
+  "c0-R2wvW1yRnRjQdUqetgD9tDJSCGpeJjz1HthfXwQ8",
+  "wsT2snFHYQ7AX7OxnrFViyu4v5il6sIb9EYxTnBnMQc",
+  "FyQ9uMx1XevItG1kE65BMvbbqcvdOGJrC_nb-PPIawk",
+  "xswbZRtkjQQ8D1h6tx503iLaAxxPLWP10J2TvgbRZXk",
+  "NQy9H6oAE-m55BheXbGu70nEWiiGMsL8lM9YsNJ8gD4",
+  "gkcnuAZeFeqPvFvNABFKGRKGE_AsmA0T3I1_jOFF0MU",
+]);
+
+/**
+ * Manual claimable token FLP IDs that require manual intervention for claiming rewards.
+ * These tokens do not support auto-claim functionality and must be claimed manually by users.
+ */
+const MANUAL_CLAIMABLE_FLP_IDS = new Set([
+  "NXZjrPKh-fQx8BUCG_OXBUtB4Ix8Xf0gbUtREFoWQ2Q", // ACTION Token
+  "rW7h9J9jE2Xp36y4SKn2HgZaOuzRmbMfBRPwrFFifHE", // AR.IO Token
+  "3eZ6_ry6FD9CB58ImCQs6Qx_rJdDUGhz-D2W1AqzHD8", // PIXL Token
+  "Wc8Rg-owsWSvrmb5XAlmSs3_4UtHo9i5ui2o9UCFuTk", // Protocol Land
+]);
+
+const CACHE_KEY = "flp-tokens";
+const CACHE_EXPIRY = 86400; // 24 hours in seconds
```

Comment:
@7i7o should we cache for 24hrs or how frequently should we refresh flp tokens data ?